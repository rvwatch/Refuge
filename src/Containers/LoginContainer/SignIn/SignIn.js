import React, { Component } from 'react';
import { connect } from 'react-redux';
import * as Actions from '../../../Actions/';
import { postLogin } from '../../../ApiCalls/postLogin';
import { object, func } from 'prop-types';
import { withRouter } from 'react-router-dom';

export class SignIn extends Component {
  constructor() {
    super();
    this.state = {
      username: '',
      password: ''
    };
  }

  handleInput = event => {
    const { name, value } = event.target;
    this.setState({
      [name]: value
    });
  };

  signInUser = async event => {
    event.preventDefault();
    const { username, password } = this.state;
    const user = await postLogin(username, password);
    this.props.addUser(user);
    this.props.setLoggedIn();
    this.props.history.push('/');
    if (user.error) {
      throw new Error('Login error');
    }
  };

  render() {
    return (
      <form className="signin-form" onSubmit={this.signInUser}>
        <h2>Sign-In</h2>
        <div>
          <input
            onChange={this.handleInput}
            type="text"
            placeholder="username"
            name="username"
            value={this.state.username}
          />
        </div>
        <div>
          <input
            onChange={this.handleInput}
            type="password"
            placeholder="password"
            name="password"
            value={this.state.password}
          />
        </div>
        <div>
          <input className="submit-btn" type="submit" value="enter" />
        </div>
      </form>
    );
  }
}

export const mapStateToProps = ({ user }) => ({ user });

export const mapDispatchToProps = dispatch => ({
  addUser: user => dispatch(Actions.addUser(user)),
  setLoggedIn: () => dispatch(Actions.setLoggedIn()),
  addErrorMessage: error => dispatch(Actions.addErrorMessage(error))
});

SignIn.propTypes = {
  target: object,
  addUser: func,
  setLoggedIn: func,
  history: object,
  addErrorMessage: func
};

export default withRouter(connect(mapStateToProps, mapDispatchToProps)(SignIn));
