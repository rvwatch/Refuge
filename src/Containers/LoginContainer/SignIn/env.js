let env = {apiKey: 'AIzaSyAcxCaic7ghfcSvxLbWeKUed3nmFNXaiME',
  authDomain: 'refuge-d2418.firebaseapp.com',
  projectId: 'refuge-d2418',
  storageBucket: 'refuge-d2418.appspot.com',
  messagingSenderId: '115368666273',
  appId: '1:115368666273:web:a7e840b2a34527ee9e3ce0',
  measurementId: 'G-S48FW6R2BT'
};

export default env;