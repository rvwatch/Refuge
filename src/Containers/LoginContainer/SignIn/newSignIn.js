import React, { useEffect, useState } from 'react';
import firebase from 'firebase';
import env from './env.js';
import '../LoginContainer.css';
// import { postLogin } from '../../../ApiCalls/postLogin';
// import { useDispatch } from 'react-redux';
import { useHistory } from "react-router-dom";
// import * as Actions from '../../../Actions/';

const NewSignIn = () => {
  useEffect( () => {
    firebase.initializeApp({
      apiKey: env.apiKey,
      authDomain: env.authDomain,
      projectId: env.projectId,
      storageBucket: env.storageBucket,
      messagingSenderId: env.messagingSenderId,
      appId: env.appId,
      measurementId: env.measurementId
    });
  }, []);
  let history = useHistory();

  const [input, setInput] = useState({});
  const [loggedIn, setLoggedIn] = useState(false);
  const [error, setError] = useState('');

  const handleInputChange = (event) => 
    setInput({
      ...input,
      [event.currentTarget.name]: event.currentTarget.value
    });

  const signInUser = async event => {
    setError('');
    event.preventDefault();
    const { email, password } = input;
    // const tryAuth = firebase.auth().signInWithEmailAndPassword(email, password);
    // console.log('====================================');
    // console.log(tryAuth);
    // console.log('====================================');
    // const tryAuth = firebase.auth().createUserWithEmailAndPassword(email, password);
    // console.log('====================================');
    // console.log(tryAuth);
    // console.log('====================================');
    firebase.auth().signInWithEmailAndPassword(email, password)
      .catch(() => {
        firebase.auth().createUserWithEmailAndPassword(email, password)
          .catch((error) => {
            console.log('====================================');
            console.log(error.message);
            console.log('====================================');
            setError(error.message);
          });
      });
    
    setLoggedIn(current => !current);
    history.push('/');
    // const user = await postLogin(username, password);
    // this.props.addUser(user);
    // this.props.setLoggedIn();
    // this.props.history.push('/');
  };

  useEffect( () => {
    console.log(loggedIn);
  }, [loggedIn]);

  return (
    <form className="signin-form" onSubmit={signInUser}>
      <h2>Sign-In</h2>
      <div>
        <input
          onChange={handleInputChange}
          type='text'
          placeholder="email"
          name="email"
        />
      </div>
      <div>
        <input
          onChange={handleInputChange}
          type="password"
          placeholder="password"
          name="password"
        />
      </div>
      <div>
        {error}
      </div>
      <div>
        <input className="submit-btn" type="submit" value="Sign-In" />
      </div>
    </form>
  );
};
export default NewSignIn;
