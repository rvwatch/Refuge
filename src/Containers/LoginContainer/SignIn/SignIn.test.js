import React from 'react';
import {SignIn} from './SignIn';
import {shallow} from 'enzyme';

describe('SignIn', () => {

  let wrapper;
  let mockpostNewUser;
  let preventDefault;
  beforeEach(() => {
    preventDefault = jest.fn();
    mockpostNewUser = jest.fn();
    wrapper = shallow(<SignIn preventDefault={preventDefault} postNewUser={mockpostNewUser} />);
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should update state on input change', () => {
    const mockEvent = { target: { value: 'ahhhhh', name: 'name' } };
    wrapper.instance().handleInput(mockEvent);
    expect(wrapper.state('name')).toEqual('ahhhhh');
  });

  it('should add a new user', async () => {
    await wrapper.instance().signInUser();
    expect(mockpostNewUser).toHaveBeenCalled();
  });
});
