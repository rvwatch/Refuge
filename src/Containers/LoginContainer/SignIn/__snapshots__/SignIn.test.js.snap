// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignIn should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SignIn
    postNewUser={[Function]}
    preventDefault={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2>
          Sign-In
</h2>,
        <div>
          <input
                    name="username"
                    onChange={[Function]}
                    placeholder="username"
                    type="text"
                    value=""
          />
</div>,
        <div>
          <input
                    name="password"
                    onChange={[Function]}
                    placeholder="password"
                    type="password"
                    value=""
          />
</div>,
        <div>
          <input
                    className="submit-btn"
                    type="submit"
                    value="enter"
          />
</div>,
      ],
      "className": "signin-form",
      "onSubmit": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Sign-In",
        },
        "ref": null,
        "rendered": "Sign-In",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <input
            name="username"
            onChange={[Function]}
            placeholder="username"
            type="text"
            value=""
/>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "name": "username",
            "onChange": [Function],
            "placeholder": "username",
            "type": "text",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <input
            name="password"
            onChange={[Function]}
            placeholder="password"
            type="password"
            value=""
/>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "name": "password",
            "onChange": [Function],
            "placeholder": "password",
            "type": "password",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <input
            className="submit-btn"
            type="submit"
            value="enter"
/>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "submit-btn",
            "type": "submit",
            "value": "enter",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        "type": "div",
      },
    ],
    "type": "form",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2>
            Sign-In
</h2>,
          <div>
            <input
                        name="username"
                        onChange={[Function]}
                        placeholder="username"
                        type="text"
                        value=""
            />
</div>,
          <div>
            <input
                        name="password"
                        onChange={[Function]}
                        placeholder="password"
                        type="password"
                        value=""
            />
</div>,
          <div>
            <input
                        className="submit-btn"
                        type="submit"
                        value="enter"
            />
</div>,
        ],
        "className": "signin-form",
        "onSubmit": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Sign-In",
          },
          "ref": null,
          "rendered": "Sign-In",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <input
              name="username"
              onChange={[Function]}
              placeholder="username"
              type="text"
              value=""
/>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "name": "username",
              "onChange": [Function],
              "placeholder": "username",
              "type": "text",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <input
              name="password"
              onChange={[Function]}
              placeholder="password"
              type="password"
              value=""
/>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "name": "password",
              "onChange": [Function],
              "placeholder": "password",
              "type": "password",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <input
              className="submit-btn"
              type="submit"
              value="enter"
/>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "submit-btn",
              "type": "submit",
              "value": "enter",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          "type": "div",
        },
      ],
      "type": "form",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
